src/common/components/rangeDropdown/RangeDropdown.tsx

```diff
@@ -64,57 +64,60 @@ const RangeDropdown: React.FC<RangeDropdownProps> = ({
   const dropdown = React.useRef<HTMLDivElement | null>(null);
   const toggleButtonRef = React.useRef<HTMLButtonElement | null>(null);
 
-  //set values without validation
+  //set values with prevalidation, rest of validation on blur
   const handleInputChange = (inputType: RANGE_INPUT, val: string) => {
-    switch (inputType) {
-      case RANGE_INPUT.MIN:
-        onChange(val, maxInputValue);
-        break;
-      case RANGE_INPUT.MAX:
-        onChange(minInputValue, val);
-        break;
+    const allowChange =
+      (maxInputEndValue && val.length <= maxInputEndValue.length) ?? true;
+    if (allowChange) {
+      switch (inputType) {
+        case RANGE_INPUT.MIN:
+          onChange(val, maxInputValue);
+          break;
+        case RANGE_INPUT.MAX:
+          onChange(minInputValue, val);
+          break;
+      }
     }
   };
 
   //validate values on blur
   const handleInputBlur = (inputType: RANGE_INPUT, val: string) => {
-    const getValidatedMinValue = (val: string): string => {
+    const getValidatedValue = (val: string): string => {
       let resultValue = val;
       if (Number(val) < Number(minInputStartValue)) {
         resultValue = minInputStartValue;
       } else if (Number(val) > Number(maxInputEndValue)) {
         resultValue = maxInputEndValue;
       }
-      if (val && maxInputValue && Number(val) > Number(maxInputValue)) {
+      if (
+        inputType !== RANGE_INPUT.MAX &&
```

Comment:
What if `inputType === RANGE_INPUT.ALL` ?

Reply:
No, its actually for RANGE_INPUT.MIN method.. so if min value > max value we fix the min value. Because its focused out, we can change only this field, and not control another field. ALL will just execute both min and max checks on component load.

Reply:
What I meant is that when `inputType === RANGE_INPUT.ALL` it also goes to this if-clause.

Reply:
can you paste the code please